subsys/net/lib/lwm2m/lwm2m_obj_connmon.c

```diff
@@ -95,7 +106,11 @@ static struct lwm2m_engine_obj_field fields[] = {
 	OBJ_FIELD_DATA(CONNMON_APN, R_OPT, STRING),
 	OBJ_FIELD_DATA(CONNMON_CELLID, R_OPT, U32),
 	OBJ_FIELD_DATA(CONNMON_SMNC, R_OPT, U16),
-	OBJ_FIELD_DATA(CONNMON_SMCC, R_OPT, U16)
+	OBJ_FIELD_DATA(CONNMON_SMCC, R_OPT, U16),
+#if defined(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2)
+	OBJ_FIELD_DATA(CONNMON_SIGNAL_SNR, R_OPT, S32),
+	OBJ_FIELD_DATA(CONNMON_LAC, R_OPT, S32),
```

Comment:
You declare a `uint16_t` buffer for this resource, the resource definition should match that choice (so either set it to `U16`, or create an `int32_t` buffer.

Reply:
This was actually also mismatch in the specifications as well. OMA Specification states that this is an integer, nothing else but also refers to the 3GPP specification which states that this is a field spanning 2 octets making it U16. I'll change the resource definition.